scripts/pre_commit_linter.py

```diff
@@ -630,1300 +876,1143 @@ def _lint_py_files(config_pylint, config_pycodestyle, files_to_lint, result):
-        parent_dir, 'node_modules', 'eslint', 'bin', 'eslint.js')
-    stylelint_path = os.path.join(
-        parent_dir, 'node_modules', 'stylelint', 'bin', 'stylelint.js')
-    config_path_for_css_in_html = os.path.join(
-        parent_dir, 'oppia', '.stylelintrc')
-    config_path_for_oppia_css = os.path.join(
-        parent_dir, 'oppia', 'core', 'templates', 'dev', 'head',
-        'css', '.stylelintrc')
-    if not (os.path.exists(eslint_path) and os.path.exists(stylelint_path)):
-        print ''
-        print 'ERROR    Please run start.sh first to install node-eslint '
-        print '         or node-stylelint and its dependencies.'
-        sys.exit(1)
+        with _redirect_stdout(_TARGET_STDOUT):
+            print '\n'.join(summary_messages)
+            print ''
 
-    js_files_to_lint = [
-        filename for filename in all_files if filename.endswith('.js')]
-    py_files_to_lint = [
-        filename for filename in all_files if filename.endswith('.py')]
-    html_files_to_lint_for_css = [
-        filename for filename in all_files if filename.endswith('.html')]
-    css_files_to_lint = [
-        filename for filename in all_files if filename.endswith('oppia.css')]
-
-    css_in_html_result = multiprocessing.Queue()
-    css_in_html_stdout = multiprocessing.Queue()
-
-    linting_processes = []
-    linting_processes.append(multiprocessing.Process(
-        target=_lint_css_files, args=(
-            node_path,
-            stylelint_path,
-            config_path_for_css_in_html,
-            html_files_to_lint_for_css, css_in_html_stdout,
-            css_in_html_result)))
-
-    css_result = multiprocessing.Queue()
-    css_stdout = multiprocessing.Queue()
-
-    linting_processes.append(multiprocessing.Process(
-        target=_lint_css_files, args=(
-            node_path,
-            stylelint_path,
-            config_path_for_oppia_css,
-            css_files_to_lint, css_stdout,
-            css_result)))
-
-    js_result = multiprocessing.Queue()
-    js_stdout = multiprocessing.Queue()
-
-    linting_processes.append(multiprocessing.Process(
-        target=_lint_js_files, args=(
-            node_path, eslint_path, js_files_to_lint,
-            js_stdout, js_result)))
-
-    py_result = multiprocessing.Queue()
-
-    linting_processes.append(multiprocessing.Process(
-        target=_lint_py_files,
-        args=(config_pylint, config_pycodestyle, py_files_to_lint, py_result)))
-
-    print 'Starting CSS, Javascript and Python Linting'
-    print '----------------------------------------'
+        return summary_messages
 
-    for process in linting_processes:
-        process.daemon = False
-        process.start()
+    def _check_directive_scope(self):
+        """This function checks that all directives have an explicit
+        scope: {} and it should not be scope: true.
+        """
+        print 'Starting directive scope check'
+        print '----------------------------------------'
```

Comment:
Shouldn't these type of print statements be covered under the verbose check? Because in the corresponding issue, @apb7 emphasizes that only the error messages be displayed. This can be printed when the verbose flag is turned on.

Reply:
Okay, will change it.

Reply:
Ok, printing only whether the check has passed or failed.

Reply:
Done